cmd/oci-sync/main.go

```diff
@@ -50,6 +52,11 @@ var flOneTime = flag.Bool("one-time", util.EnvBool("OCI_SYNC_ONE_TIME", false),
 var flMaxSyncFailures = flag.Int("max-sync-failures", util.EnvInt("OCI_SYNC_MAX_SYNC_FAILURES", 0),
 	"the number of consecutive failures allowed before aborting (the first sync must succeed, -1 will retry forever after the initial sync)")
 
+func errorBackoff() wait.Backoff {
+	durationLimit := math.Max(*flWait, float64(util.MinimumSyncContainerBackoffCap))
+	return util.BackoffWithDurationLimit(util.WaitTime(durationLimit), math.MaxInt32)
```

Comment:
We can use `*flMaxSyncFailures` as the step cap if it is not `-1`.
We'll need to update how flMaxSyncFailures is used.

Reply:
In L47, the description of the `--wait` flag is not true. Instead of a fixed wait period, it now represents the wait duration cap. We need to update it, and the helm-sync too.

Reply:
I intentionally chose not to add a step cap here, I think it just makes the backoff more complex to understand. For this use case I consider it simpler to just define either a duration cap or a step cap, but not both. Considering the process exits when `*flMaxSyncFailures` is reached, it makes no difference if we map this to the step cap.

`--wait` is still the number of seconds between syncs for the happy path. It's only used as the duration cap if the provided value is greater than the default cap of 1 hour. I think this can be treated as an implementation detail.